packages/catalog-process/src/services/catalogService.ts

```diff
@@ -658,21 +632,47 @@ async function innerAddDocumentToEserviceEvent(
     uploadDate: new Date(),
   };
 
-  const templateVersionRef = evaluateEServiceTemplateVersionRef(
-    templateVersionId,
-    documentSeed
-  );
+  function evaluateTemplateVersionRef(
+    descriptor: Descriptor
+  ): Descriptor["templateVersionRef"] {
+    if (
+      !isInterface ||
+      !descriptor.templateVersionRef ||
+      !documentSeed.interfaceTemplateMetadata
+    ) {
+      return descriptor.templateVersionRef;
+    }
+
+    const {
+      contactEmail,
+      contactName,
+      contactUrl,
+      serverUrls,
+      termsAndConditionsUrl,
+    } = documentSeed.interfaceTemplateMetadata;
+
+    return {
+      id: descriptor.templateVersionRef.id,
+      interfaceMetadata: {
+        contactEmail,
+        contactName,
+        contactUrl,
+        serverUrls,
+        termsAndConditionsUrl,
+      },
+    };
+  }
```

Comment:
I would to move this fn outside `innerAddDocumentToEserviceEvent` adding a necessary parameter instead using `documentSeed` and `isInterface` variable hoisted in previous fn

Reply:
Done!